doc/source/admin/dependency_resolvers.rst

```diff
@@ -0,0 +1,143 @@
+
+The binding between tool XML and the tools they need to run is specified in the tool XML using *requirement*
+tags, for example
+
+.. code-block:: xml
+
+    <requirement type="package" version="0.7.10.039ea20639">bwa</requirement>
+
+In some cases these requirement tags can be specified without a version
+
+.. code-block:: xml
+
+    <requirement type="package">bedtools</requirement>
+
+The requirement turn into inputs to the dependency resolver. Each dependency resolver is thus given given one or
+two inputs: the name of the dependency to resolve and, in most cases, the version number of the
+dependency. By default, the dependency resolver configuration contains three items. First, the
+*tool shed dependency resolver* is used first, then the *Galaxy packages dependency resolver* is used, first
+looking for packages by name and version number and then finally looking for the package just by name. The default
+configuration thus prefers packages installed from the Galaxy Tool Shed, before trying to find a "Galaxy package"
+satisfying the specific version the dependency requires before finally falling back to looking for a Galaxy package
+with merely the correct name.
+
+Default Dependency Resolvers
+----------------------------
+
+The default configuration of dependency resolvers is equivalent to the following ``dependency_resolvers_conf.xml``
+
+.. code-block:: xml
+
+  <dependency_resolvers>
+  <!-- the default configuration, first look for dependencies installed from the toolshed -->
+    <tool_shed_packages />
+  <!-- then look for env.sh files profile according to the "galaxy packages" schema -->
+    <galaxy_packages />
+    <galaxy_packages versionless="true" />
+  </dependency_resolvers>
+
+Tool Shed Dependency Resolver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``tool_shed_packages`` dependency resolver works with packages installed from the Galaxy Tool Shed. When a package
+is installed from the Tool Shed it creates a directory structure under the directory that is specified as the
+``tool_dependency_dir`` in Galaxy's configuration. This directory structure contains references to the tool's name,
+owner (in the Tool Shed) and version number (amongst other things) and ultimately contains a file name ``env.sh``
+that contains commands to make the dependency runnable. This is installed, along with the packaged tool, by the tool
+package and doesn't require any configuration by the Galaxy administrator.
+
+The Tool Shed dependency resolver is not able to resolve package requirements that do not have a version number,
+like the `bedtools` example above.
+
+Galaxy Packages Dependency Resolver
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``galaxy_packages`` dependency resolver is allows Galaxy admins to specify how Galaxy should load manually
+installed packages. In the default configuration this is called twice, once immediately after
+the ``tool_shed_packages`` resolver and then a second time in its *versionless* mode.
```

Comment:
Actually, in the default configuration this is called 0, 1 or 2 times, depending on the previous resolvers succeeding or not. I'd just say that "This resolver can be configured either to use the version string or in *versionless* mode.